Migrate security documentation to its own chapter (#4283)
jenkins-infra/jenkins.io: content/doc/book/security/configuring-content-security-policy.adoc

Therefore any file stored in a build directory and served by Jenkins should be considered potentially unsafe.

If either of these are true, you should be very careful when relaxing the CSP rule set.
If neither is true, and all users with the ability to change files in workspaces are fully trusted, as are all agent machines, then it should be safe to relax or even disable the CSP rules.

=== Implementation

The CSP header sent by Jenkins can be modified by setting the system property `hudson.model.DirectoryBrowserSupport.CSP`:

If its value is the *empty string*, e.g. `java -Dhudson.model.DirectoryBrowserSupport.CSP= -jar jenkins.war` then the header will not be sent at all.

WARNING: This is potentially very unsafe and should only be used after reviewing the overall security setup.

Any other value will be used as the header value, e.g. `java -Dhudson.model.DirectoryBrowserSupport.CSP="sandbox; default-src 'self';" -jar jenkins.war`.
See http://content-security-policy.com/[content-security-policy.com] for a reference on this header and its possible values.

> Changes to the system property will be effective immediately, so it's possible to set this system property temporarily via the <<../../managing/script-console#,Script Console>>, allowing you to experiment with different values:

Reviewer: Optional
```suggestion
Changes to the system property are effective immediately, so it's possible to set this system property temporarily via the <<../../managing/script-console#,Script Console>>, allowing you to experiment with different values:
```
